python/ray/dag/collective_node.py

```diff
@@ -124,8 +134,21 @@ def execute(self, send_buf: "torch.Tensor") -> "torch.Tensor":
         if not isinstance(send_buf, torch.Tensor):
             raise ValueError("Expected a torch tensor")
         communicator = self.get_communicator()
-        recv_buf = torch.empty_like(send_buf)
-        communicator.allreduce(send_buf, recv_buf, self._op)
+        if self.comm_op == self.ALLREDUCE:
+            recv_buf = torch.empty_like(send_buf)
+            communicator.allreduce(send_buf, recv_buf, self._op)
+        elif self.comm_op == self.REDUCESCATTER:
+            world_size = len(self._actor_handles)
+            assert (
+                send_buf.shape[0] % world_size == 0
+            ), "Input tensor's first dimension should be divisible by "
```

Comment:
Raise a ValueError here. For all the user-input errors, we would raise a ValueError instead of asserting. Assert is used for checking internal states.